### PR TITLE
[CELEBORN-1629] Support to apply ratis election operation with RESTful api

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?

As title, sub task of [CELEBORN-1628](https://issues.apache.org/jira/browse/CELEBORN-1628)
### Why are the changes needed?

It is more friendly for customer experience and necessary for automation integration.

image

Before restart/recreate the celeborn master pod
- call the  ratis/snapshot/create to create the snapshot(will raise another PR for this)
- if need to failover
    - call the ratis/election/pause to pause the election
    - call the ratis/election/step_down or ratis/election/transfer to transfer the leadership
    - call the ratis/election/resume to resume the election finally

### Does this PR introduce _any_ user-facing change?
Introduce new RESTful apis.

### How was this patch tested?

election transfer:
image

election pause and resume:

image

election step down:
image

### CODE REVIEW

**Comment by mridulm** [comment]
+CC @akpatnam25 FYI for CLI

**Comment by turboFei** [comment]
It is done for ratis election apis.

cc @mridulm

**Comment by turboFei** [comment]
cc @AngersZhuuuu I saw that you add the docs for ratis-shell.

also cc @waitinfuture @RexXiong

**Review by RexXiong — APPROVED** [review]
LGTM, Could you also add the usage of the Ratis REST API in https://celeborn.apache.org/docs/latest/celeborn_ratis_shell

**Comment by turboFei** [comment]
> Could you also add the usage of the Ratis REST API in https://celeborn.apache.org/docs/latest/celeborn_ratis_shell

Sure, will add it later

File: docs/celeborn_ratis_shell.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ license: |
 Celeborn uses Ratis to implement the HA function of the master, Celeborn directly introduces ratis-shell package into the project
 then it's convenient for Celeborn Admin to operate the master ratis service.
 
+Since celeborn-0.6.0, the ratis RESTful API is supported, which is more convenient to operate the ratis service, see details in the swagger: `http://<CELEBORN_HOST>:<CELEBORN_PORT>/swagger#/Ratis`.
```
Comment (turboFei) [inline docs/celeborn_ratis_shell.md:25]: Seems it is enough for the docs, cc @RexXiong 

image

Comment (turboFei) [inline docs/celeborn_ratis_shell.md:25]: I think it is enough for the docs.

image

File: docs/celeborn_ratis_shell.md [hunk 2]
```diff
@@ -22,6 +22,8 @@ license: |
 Celeborn uses Ratis to implement the HA function of the master, Celeborn directly introduces ratis-shell package into the project
 then it's convenient for Celeborn Admin to operate the master ratis service.
 
+Since celeborn-0.6.0, the ratis [RESTful API](webapi.md) is supported, which is more convenient to operate the ratis service, see details in the swagger: `http://<CELEBORN_HOST>:<CELEBORN_PORT>/swagger#/Ratis`.
```
Comment (turboFei) [inline docs/celeborn_ratis_shell.md:25]: I think it is enough for the docs.

image

File: docs/celeborn_ratis_shell.md [hunk 3]
```diff
@@ -22,6 +22,8 @@ license: |
 Celeborn uses Ratis to implement the HA function of the master, Celeborn directly introduces ratis-shell package into the project
 then it's convenient for Celeborn Admin to operate the master ratis service.
 
+Since celeborn-0.6.0, the ratis [RESTful API](webapi.md) is supported, which is more convenient to operate the ratis service, see details in the swagger: `http://<CELEBORN_HOST>:<CELEBORN_PORT>/swagger/#/Ratis`.
```
Comment (turboFei) [inline docs/celeborn_ratis_shell.md:25]: I think it is enough for the docs.

image

File: docs/celeborn_ratis_shell.md [hunk 4]
```diff
@@ -22,6 +22,8 @@ license: |
 Celeborn uses Ratis to implement the HA function of the master, Celeborn directly introduces ratis-shell package into the project
 then it's convenient for Celeborn Admin to operate the master ratis service.
 
+Since 0.6.0, the ratis [RESTful API](webapi.md) is supported, which is more convenient to operate the ratis service, see details in the swagger: `http://<CELEBORN_HOST>:<CELEBORN_PORT>/swagger/#/Ratis`.
```
Comment (turboFei) [inline docs/celeborn_ratis_shell.md:25]: I think it is enough for the docs.

image

Comment (RexXiong) [reply docs/celeborn_ratis_shell.md:25]: > I think it is enough for the docs.
> 
> image

Sure

**Comment by RexXiong** [comment]
Thanks. merge to main(v0.6.0)

**Pull request closed without merging**